client/src/Tickets/TicketModal.jsx

```diff
@@ -84,26 +86,78 @@ function Charger({ stateChange }) {
   );
 }
 
-const AddTicketModel = ({ update, data, stateChange, toggleUserModal, setToggleUserModal }) => {
+const TicketModal = ({ onCreate, onUpdate }) => {
+  const navigate = useNavigate();
+  const { ticketId } = useParams();
+  const [data, setData] = useState({
+    ClientId: null,
+    LocationId: null,
+    UserId: null,
+    serialNumber: '',
+    device: '',
+    problem: '',
+    troubleshooting: '',
+    resolution: '',
+    dateOn: DateTime.now().toISODate(),
+    timeInAt: '',
+    timeOutAt: '',
+    hasCharger: false,
+    notes: '',
+  });
```

Comment:
Notice that the data for the modal is now declared with useState INSIDE THE MODAL, not passed in from the parent.

Reply:
Hence, the modal "encapsulates" the data that it manages...